changelogs/unreleased/gh-8899-compare-with-key-slowpath-invalid-memory-access.md

```diff
@@ -0,0 +1,4 @@
+## bugfix/box
+
+* Fixed the invalid memory access in a corner case of a specialized comparison
+  function (gh-8899).
```

Comment:
```suggestion
* Fixed an invalid memory access in a corner case of a specific comparison
  function (gh-8899).
```

Or maybe mention the function name explicitly, if it is visible to users.

Reply:
Hi, fixed the article. Can you please verify the new changelog?

The comparison function is specialized on non-sequential keys and keys with JSON paths.

This is why the function is called "specialized" - it only works with the keys satisfying its requirements.

Reply:
I bet everything is fine here.
Actually this is backport to 2.10, so we have already passed doc check when pushed #8922.